docs/src/glossary.md

```diff
@@ -0,0 +1,85 @@
+# ChainRules Glossary
+
+This glossary serves as a quick reference for common terms used in the field of Automatic Differentiation, as well as those used throughout the documentation relating specifically to ChainRules.
+
+##Definitions:
+
+###Adjoint:
+
+The conjugate transpose of the Jacobian for a given function `f`.
```

Comment:
Probably we want to title the Adjoint of a function.
ANd start by saying that the adjoint of a matrix is another word for it's conjugate transpose.
Then mentioning that it can also be applied to a linear operator as every linear operator can be described as `y = Jx` and that as an adjoint linear operator of `y' = x'J'`.
Then say that people say as a shorthand/abuse of terminoly the adjoint of a function,
when what they actually mean is to get a function which is the adjoint of pushfoward linear operator.
The pushforward linear operator is the the linear operator that has the same jacobian as the function at that point.
The pullback is its adjoint.
 linearization of the function at a point, to get a linear operator (the `pushforward`),
and then 
Then say that people occationally say the adjoint of a function,
when what they really mean is: the adjoint of the jacobian of the function,
or they mean the pullback.
Sometimes people say adjoint of a function to mean pullback.

Reply:
The adjoint is super broad as a term, so I am having a hard time figuring out how much / how little to include